internal/service/fw/model_atcfw_application_filter.go

```diff
@@ -0,0 +1,149 @@
+	"policies":     types.ListType{ElemType: types.StringType},
+	"readonly":     types.BoolType,
+	"tags":         types.MapType{ElemType: types.StringType},
+	"updated_time": timetypes.RFC3339Type{},
+}
+
+var AtcfwApplicationFilterResourceSchemaAttributes = map[string]schema.Attribute{
+	"created_time": schema.StringAttribute{
+		CustomType:          timetypes.RFC3339Type{},
+		Computed:            true,
+		MarkdownDescription: "The time when this Application Filter object was created.",
+	},
+	"criteria": schema.ListNestedAttribute{
+		NestedObject: schema.NestedAttributeObject{
+			Attributes: AtcfwApplicationCriterionResourceSchemaAttributes,
+		},
+		Required:            true,
+		MarkdownDescription: "The array of key-value pairs specifying criteria for the search.",
+	},
+	"description": schema.StringAttribute{
+		Optional:            true,
+		Computed:            true,
+		Default:             stringdefault.StaticString(""),
+		MarkdownDescription: "The brief description for the application filter.",
+	},
+	"id": schema.Int64Attribute{
+		Computed: true,
+		PlanModifiers: []planmodifier.Int64{
+			int64planmodifier.UseStateForUnknown(),
+		},
+		MarkdownDescription: "The Application Filter object identifier.",
+	},
+	"name": schema.StringAttribute{
+		Required:            true,
+		MarkdownDescription: "The name of the application filter.",
+	},
+	"policies": schema.ListAttribute{
+		ElementType:         types.StringType,
+		Computed:            true,
+		MarkdownDescription: "The list of security policy names with which the application filter is associated.",
+	},
+	"readonly": schema.BoolAttribute{
+		Computed:            true,
+		MarkdownDescription: "True if it is a predefined application filter",
+	},
+	"tags": schema.MapAttribute{
+		ElementType:         types.StringType,
+		Optional:            true,
+		MarkdownDescription: "Enables tag support for resource where tags attribute contains user-defined key value pairs",
+	},
+	"updated_time": schema.StringAttribute{
+		CustomType:          timetypes.RFC3339Type{},
+		Computed:            true,
+		MarkdownDescription: "The time when this Application Filter object was last updated.",
+	},
+}
+
+func ExpandAtcfwApplicationFilter(ctx context.Context, o types.Object, diags *diag.Diagnostics) *fw.ApplicationFilter {
+	if o.IsNull() || o.IsUnknown() {
+		return nil
+	}
+	var m AtcfwApplicationFilterModel
+	diags.Append(o.As(ctx, &m, basetypes.ObjectAsOptions{})...)
+	if diags.HasError() {
+		return nil
+	}
+	return m.Expand(ctx, diags)
+}
+
+func (m *AtcfwApplicationFilterModel) Expand(ctx context.Context, diags *diag.Diagnostics) *fw.ApplicationFilter {
+	if m == nil {
+		return nil
+	}
+	to := &fw.ApplicationFilter{
+		Criteria:    flex.ExpandFrameworkListNestedBlock(ctx, m.Criteria, diags, ExpandAtcfwApplicationCriterion),
+		Description: flex.ExpandStringPointer(m.Description),
+		Name:        flex.ExpandStringPointer(m.Name),
+		Readonly:    flex.ExpandBoolPointer(m.Readonly),
```

Comment:
if it is "Computed/Read only" it shouldn't be Expanded.

Reply:
Fixed